packages/react-juce/src/components/ListView.tsx

```diff
@@ -0,0 +1,218 @@
+  _calculateVirtualPositions(): VirtualPositions {
+    const totalItems  = this.props.data.length;
+    const innerHeight = this.props.itemHeight * totalItems;
+
+    invariant(
+      this.props.itemHeight > 0,
+      'Zero or negative itemHeight passed to ListView'
+    );
+
+    // Pad num rendered items by 1 so we always render the same number of items
+    // which allows us to use a fixed key value on each item in the list.
+    const numItems = Math.floor(this.state.height / this.props.itemHeight) + 1;
+
+    const startIndex = Math.floor(this.state.scrollTopPosition / this.props.itemHeight);
+    const endIndex   = Math.min(totalItems - 1, Math.floor(startIndex + numItems));
+
+    return {
+      innerHeight: innerHeight,
+      startIndex: startIndex,
+      endIndex: endIndex
+    }
+  }
+
+  _setScrollTopPosition(e: ScrollEvent): void {
+    this.setState({
+      scrollTopPosition: e.scrollTop
+    });
+
+    if (typeof this.props.onScroll === 'function') {
+      this.props.onScroll(e);
+    }
+  }
+
+  scrollToIndex(scrollParams: ScrollParams) {
+    invariant(scrollParams.index >= 0 && scrollParams.index <= this.props.data.length,
+              "scrollParams.index must be between 0 and props.data.length");
+
+    invariant(scrollParams.offset >= 0.0 && scrollParams.offset <= 1.0,
+              "scrollParams.offset must be normalised between 0.0 and 1.0");
+
+    const numItems = Math.floor(this.state.height / this.props.itemHeight);
+
+    const xPos = 0;
+    let   yPos = (this.props.itemHeight * scrollParams.index) -
+                 (this.props.itemHeight * numItems * scrollParams.offset);
+
+    const scrollViewInstance = this._ref ? this._ref.current : null;
+
+    if (scrollViewInstance) {
+      //@ts-ignore
+      scrollViewInstance.scrollToPosition(xPos, yPos);
+    }
+  }
+
+  render() {
+    const positions: VirtualPositions = this._calculateVirtualPositions();
+    const items: any = [];
+
+    // List items must have a key but we ensure the key remains fixed here
+    // so that we simply re-render the existing views/components rather than
+    // replacing them. This stops issues occuring when dynamically adding and removing
+    // components inside a juce::Viewport.
+    for (let i = positions.startIndex; i <= positions.endIndex; ++i) {
+      items.push(
+        this.props.renderItem(
+          this.props.data[i],
+          i,
+          {
+            position: 'absolute',
+            top: this.props.itemHeight * i,
+            key: positions.endIndex - i,
+          }
+        )
+      );
+    }
+
+    return (
+      <ScrollView
+        {...this.props}
+        onMeasure={this._onMeasure}
+        onScroll={this._setScrollTopPosition}
+        viewRef={this._ref}
+      >
+        <ScrollView.ContentView {...styles.scroll_view_content} height={positions.innerHeight}>
+          {items}
+        </ScrollView.ContentView>
+      </ScrollView>
+    )
+  }
+}
+
+const styles = {
+  scroll_view_content: {
```

Comment:
Major nitpick: camelCase please. Once we get `prettier` involved we'll catch these things automatically